core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -214,13 +215,11 @@ public final Solution_ solve(Solution_ problem) {
             } finally {
                 sample.stop();
                 Metrics.globalRegistry.remove(new Meter.Id(SolverMetric.SCORE_CALCULATION_COUNT.getMeterId(),
-                        solverScope.getMonitoringTags(),
-                        null,
-                        null,
-                        Meter.Type.GAUGE));
+                        solverScope.getMonitoringTags(), null, null, Meter.Type.GAUGE));
                 solverScope.getSolverMetricSet().forEach(solverMetric -> solverMetric.unregister(this));
             }
             restartSolver = checkProblemFactChanges();
+            warmStart = restartSolver;
```

Comment:
Technically, `warmStart = true` here. The `warmStart` is something that I like the least about this PR but I don't have a specific suggestion for improvement here. My gut feeling is that it should be hidden inside the solver scope. Feel free to keep it like this.